doc/mgmt/gnmi/gnsi.md

```diff
@@ -0,0 +1,383 @@
+
+Certz.Rotate() will permit rotation, and verification of function, of any of the PKI elements. The normal use-case would be to:
+- send a PKI element to a network system as a RotateCertificateRequest.
+- verify that the services which will use the new certificate bundle continue to operate normally.
+- send a FinalizeRequest to finish the rotation process.
+
+The Certz Rotate Flow Diagram
+![Rotate Flow Chart](images/gnsi_certz_rotate.png)
+
+##### Profiles
+
+SSL profiles logically group the PKI primitives together and any gRPC Server may be configured to use a given profile. There is a default profile with the `gnxi` id that gNMI/gNOI/gNSI use. If no profile is specified, this is the profile that will be referenced.
+
+There are 3 RPC for profiles:
+- Certz.GetProfileList() - Existing profiles on a target can be discovered.
+- Certz.AddProfile() - An SSL profile can be added.
+- Certz.DeleteProfile() - When a profile is no longer needed it can be removed. Note that the gnxi SSL profile cannot be removed.
+
+##### Certificate Signing Request
+
+The server can optionally support generating a Certificate Signing Request. This allows the client to open a Certz.Rotate() stream, request a CSR and then have the Certificate signed externally.
+
+- Certz.CanGenerateCSR()
+- Certz.Rotate(GenerateCSRRequest)
+
+#### Authz
+Authz is policy based authorization for gRPC access. The policy is enforced by the gRPC server by registering a file watcher with a gRPC Interceptor in the options. The policy is managed by the gNSI Authz service running in the gNMI server. \
+
+The policy to be enforced is defined in the form of a JSON string whose schema can be found [here](https://github.com/grpc/proposal/blob/master/A43-grpc-authorization-api.md).
+
+- Authz.Rotate()
+  - upload a new policy - same flow as Certz diagram
+- Authz.Probe()
+  - test the current policy
+
+#### Pathz
+
+Pathz is policy based authorization for gNMI access that is tunable for read/write at any level in the tree. The policy must be enforced by the gnmi server. A policy processor will be added as a new library to authorize gnmi requests. A call to the policy processor will be placed at the start of any gNMI request.
+
+- Pathz.Rotate()
+  - upload a new policy - same flow as Certz diagram
+- Pathz.Probe()
+  - test the current policy
+
+#### Credentialz
+
+Credentialz is for managing SSH and Console access. Changes are made through dbus calls to appropiate host service modules. The expected calls and behavior of the host service modules are defined in the next sections.
+
+- Credentialz.RotateAccountCredentials()
+  - Modify users' keys, passwords, and certificates
+- Credentialz.RotateHostParameters()
+  - Modify host's keys and certificates
+
+##### Console Mgmt (users/passwords)
```

Comment:
I assume this is the flow of Credentialz.RotateAccountCredentials(). If so, can we state this explicitly to avoid confusion for someone not familiar. Same for below.

Also does the list below (Create checkpoint, Restore checkpoint...) has any particular order? Or is it just a set of API that can be invoked in any order (besides you have to create before restore, etc).

If it is the former, we should put "set credential" at proper location.

Same for the ssh part.

Reply:
Added further explanation